libbindgen/src/lib.rs

```diff
@@ -624,14 +624,20 @@ pub fn parse_one(ctx: &mut BindgenContext,
 }
 
 /// Parse the Clang AST into our `Item` internal representation.
-fn parse(context: &mut BindgenContext) {
+fn parse(context: &mut BindgenContext) -> Result<(), ()> {
     use clang::Diagnostic;
     use clangll::*;
 
+    let mut any_error = false;
```

Comment:
I think it would be easier to get `context.translation_unit().diags()` first, and if it is not empty, iterate it and return `Err`.